### PR TITLE
Add boot options for creating and restorinig backups

### PR DESCRIPTION
**Pull request by kmorrison**
TODO: I'm not sure if we have to stop the backup engine on server shutdown the same way the cleaner does. In fact I'm not sure why we have to stop the cleaner even, when the server is dying.

NOTE: This is branched off @ogrubes db keys branch since master is currently working with corrupted options files that don't play well with the backup engine.

NOTE: Zero unit tests. Uh.... not really sure how to test this stuff, since valid tests probably involve deleting databases and I'm not sure of an easy way to do that. Advice solicited.

### CODE REVIEW

File: storage/backup.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			if err != nil {
+				log.Errorf("Automatic backup failed with error %v", err)
+			}
+		case <-backupCreator.stopChan:
+			return
+		}
+	}
+}
+
+func DoABackup(projectDB *DB, backupDir string) error {
```
Comment (kriswehner) [inline storage/backup.go]: How about making this a private method like "performBackup" on the BackupCreator?

**Comment by kriswehner** [comment]
LGTM!

File: storage/backup.go (hunk 1 above)
Comment (kmorrison) [inline storage/backup.go]: I considered doing this. However, I found it nicer, especially when testing, to try to make this a pure function. Do you feel strongly?

Comment (kriswehner) [inline storage/backup.go]: I do not, but could we rename it? Maybe "PerformBackup"?

**Pull request merged**